### PR TITLE
[14.0][MIG] product_supplierinfo_revision: Migration to 14.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 14.0

Please @joao-p-marques and @CarlosRoca13 can you review it?

@Tecnativa TT31817

### CODE REVIEW

**Review by CarlosRoca13 — APPROVED** [review]

File: product_supplierinfo_revision/wizards/supplierinfo_duplicate_wizard.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    date_end = fields.Date()
+    variation_percent = fields.Float(
+        digits="Product Price",
+        string="Variation %",
+    )
+
+    def action_apply(self):
+        Supplierinfo = self.env["product.supplierinfo"]
+        supplierinfo_news = Supplierinfo
+        for item in Supplierinfo.browse(self.env.context["active_ids"]):
```
Comment (rousseldenis) [inline product_supplierinfo_revision/wizards/supplierinfo_duplicate_wizard.py]:
```suggestion
        for item in Supplierinfo.browse(self.env.context.get("active_ids")):
```

File: product_supplierinfo_revision/wizards/supplierinfo_duplicate_wizard.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                    "date_start": self.date_start,
+                    "date_end": self.date_end,
+                    "previous_info_id": item.id,
+                    "price": item.price * (1.0 + self.variation_percent / 100.0),
+                }
+            )
+            item.date_end = fields.Date.from_string(self.date_start) - relativedelta(
+                days=1
+            )
+        action = self.env.ref("product.product_supplierinfo_type_action").read()[0]
```
Comment (rousseldenis) [inline product_supplierinfo_revision/wizards/supplierinfo_duplicate_wizard.py]:
```suggestion
        action = self.env["ir.actions.actions"]._for_xml_id("product.product_supplierinfo_type_action")
```

File: product_supplierinfo_revision/wizards/supplierinfo_duplicate_wizard.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    variation_percent = fields.Float(
+        digits="Product Price",
+        string="Variation %",
+    )
+
+    def action_apply(self):
+        Supplierinfo = self.env["product.supplierinfo"]
+        supplierinfo_news = Supplierinfo
+        for item in Supplierinfo.browse(self.env.context["active_ids"]):
+            supplierinfo_news |= item.copy(
```
Comment (rousseldenis) [inline product_supplierinfo_revision/wizards/supplierinfo_duplicate_wizard.py]: I suggest to use ids list instead of recordset for performances reasons.

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

**Review by joao-p-marques — APPROVED** [review]
Code review. LGTM :+1:

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**